### DISCUSSION TITLE
After 'apply refresh policy' only Incremental periods are refreshed?

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [TabularEditor/TabularEditor3#1348] [🙋♀️TE3 Q&A] [answered]
After publishing PBIX if you use 'apply refresh policy' of TE3 and do a full refresh then only incremental periods are refreshed older periods are ignored and have to be refreshed manually this is different behaviour than the refresh option of PBI Service where all the periods are refreshed in the first refresh and later only incremental periods are refreshed any reason for that?

**Comment by otykier** [comment, marked as answer]
Well, Tabular Editor was not designed to mimic every aspect of Power BI. In particular, we try to avoid triggering automatic operations that could potentially take a long time to complete, or cause side-effects such as clearing data from table partitions. For this reason, the “apply refresh policy” does exactly what its [documentation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.analysisservices.tabular.table.applyrefreshpolicy?view=analysisservices-dotnet) says, and nothing else: That is, it creates/deletes/merges partitions based on the refresh policy. If some or all partitions needs to be refreshed after doing this, we trust the user to perform this refresh afterwards. There could also be good reasons for not wanting to perform the refresh immediately after applying the refresh policy.
